src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -26,4 +31,21 @@
         return userChallengeList.stream().map(userChallenge -> UserChallengeMapper.convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
 
     }
+
+    @Override
+    public UserChallenge rateUser(UserChallengeDTO userChallengeDTO, Long ownerId, Long userChallengeId) {
+
+        userChallengeDTO.setId(userChallengeId);
+        UserChallenge userChallenge = userChallengeRepository.findById(userChallengeDTO.getId()).get();
+        userChallenge.setPoints(userChallengeDTO.getPoints());
+
+        Challenge challenge = challengeRepository.findById(userChallenge.getChallenge().getId()).get();
```

Comment:
Si aici si mai sus, inainte de get cred ca e bine sa verificam daca ce citim din baza de date exista in ea si daca nu sa aruncam exceptii. Este o metoda care se poate apela pe Optional<T>, isPresent() se numeste.